File: packages/swift/Sources/CatalystSwift/Models/CatalystArchive.swift
// Licensed under the MIT License
//
// Copyright (c) 2025 Natsune Mochizuki

/// A single month bucket in the status archive.
public struct CatalystArchiveMonth: Decodable, Sendable, Equatable, Hashable {
  public let year: Int
  public let month: Int
  public let count: Int
}

public struct CatalystArchiveMonths: Decodable, Sendable {
  public let months: [CatalystArchiveMonth]
}

File: packages/kotlin/src/main/kotlin/com/natsuneko/catalyst/models/CatalystArchive.kt
// Licensed under the MIT License
//
// Copyright (c) 2025 Natsune Mochizuki

package com.natsuneko.catalyst.models

import kotlinx.serialization.Serializable

/**
 * A single month bucket in the status archive
 */
@Serializable
data class CatalystArchiveMonth(
    val year: Int,
    val month: Int,
    val count: Int
)

/**
 * Wrapper for the `{"months": [...]}` response shape
 */
@Serializable
internal data class CatalystArchiveMonths(val months: List<CatalystArchiveMonth>)
